js/ui/grid_core/ui.grid_core.focus.js

```diff
@@ -252,27 +252,29 @@ const FocusController = core.ViewController.inherit((function() {
             }
         },
         _navigateToVirtualRow: function(key, deferred, needFocusRow) {
-            const that = this;
             const dataController = this.getController('data');
             // @ts-expect-error
             const rowsScrollController = dataController._rowsScrollController;
             const rowIndex = gridCoreUtils.getIndexByKey(key, dataController.items(true));
-            const scrollable = that.getView('rowsView').getScrollable();
+            const scrollable = this.getView('rowsView').getScrollable();
 
             if(rowsScrollController && scrollable && rowIndex >= 0) {
                 const focusedRowIndex = rowIndex + dataController.getRowIndexOffset(true);
                 const offset = rowsScrollController.getItemOffset(focusedRowIndex);
 
-                const triggerUpdateFocusedRow = function() {
-                    that.component.off('contentReady', triggerUpdateFocusedRow);
+                const triggerUpdateFocusedRow = () => {
+                    if(dataController.totalCount() && !dataController.items().length) {
```

Comment:
Minor comment.

We can use here ```dataController.isEmpty()```:
```suggestion
                    if(dataController.totalCount() && dataController.isEmpty()) {
```